src/macaron/slsa_analyzer/package_registry/deps_dev.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2025 - 2025, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""This module contains implementation of deps.dev service."""
+
+import json
+import logging
+import urllib.parse
+from json.decoder import JSONDecodeError
+from urllib.parse import quote as encode
+
+from macaron.config.defaults import defaults
+from macaron.errors import ConfigurationError, InvalidHTTPResponseError
+from macaron.util import send_get_http_raw
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class DepsDevService:
+    """The deps.dev service class."""
+
+    @staticmethod
+    def get_package_info(purl: str) -> dict | None:
+        """Check if the package identified by the PackageURL (PURL) exists and return its information.
+
+        Parameters
+        ----------
+        purl: str
+            The PackageURL (PURL).
+
+        Returns
+        -------
+        dict | None
+            The package metadata or None if it doesn't exist.
+
+        Raises
+        ------
+        InvalidHTTPResponseError
+            If a network error happens or unexpected response is returned by the API.
+        """
+        section_name = "deps_dev"
+        if not defaults.has_section(section_name):
+            return None
+        section = defaults[section_name]
+
+        url_netloc = section.get("url_netloc")
+        if not url_netloc:
+            raise ConfigurationError(
```

Comment:
This parts raises `ConfigurationError` but the doc string of this method doesn't include this as a possibly raised exception.

Reply:
I have created a more generic error, `APIAccessError` to account for different cases. 

See commit [20155c3](https://github.com/oracle/macaron/pull/956/commits/20155c37d9d4709f64a7a26eb2ca1e231f41feaa).